pytorch_lightning/core/lightning.py

```diff
@@ -1729,6 +1730,33 @@ def to_onnx(self, file_path: str, input_sample: Optional[Tensor] = None, **kwarg
 
         torch.onnx.export(self, input_data, file_path, **kwargs)
 
+    def to_torchscript(self) -> torch.jit.ScriptModule:
+        """
+        Compiles the model to a :class:`~torch.jit.ScriptModule`.
+        This can be overridden to support custom TorchScript module export.
+
+        Note:
+            Requires the implementation of the :meth:`LightningModule.forward` method.
+
+        Example:
+            >>> class SimpleModel(LightningModule):
+            ...     def __init__(self):
+            ...         super().__init__()
+            ...         self.l1 = torch.nn.Linear(in_features=64, out_features=4)
+            ...
+            ...     def forward(self, x):
+            ...         return torch.relu(self.l1(x.view(x.size(0), -1)))
+            ...
+            >>> model = SimpleModel()
+            >>> torch.jit.save(model.to_torchscript(), "model.jit")
+            >>> os.path.isfile("model.jit")
+            True
+        """
+        mode = self.training
+        scripted_module = torch.jit.script(self.eval())
```

Comment:
Wrap the script call with `torch.no_grad()` ?

Reply:
good idea! currently working on tests.